src/download.test.ts

```diff
@@ -42,23 +35,20 @@ Object.defineProperty(window, "location", {
 });
 
 describe("downloadFile", () => {
-  it("should download with a skylink url with attachment set", () => {
-    validSkylinkVariations.forEach(([fullSkylink, path]) => {
-      mockLocationAssign.mockClear();
-      const url = client.downloadFile(fullSkylink);
-
-      let expectedUrl2 = `${expectedUrl}${path}${attachment}`;
-      if (path.startsWith("#")) {
-        // Hash should come after query.
-        expectedUrl2 = `${expectedUrl}${attachment}${path}`;
-      }
-      // Change ?attachment=true to &attachment=true if need be.
-      if ((expectedUrl2.match(/\?/g) || []).length > 1) {
-        expectedUrl2 = expectedUrl2.replace(attachment, "&attachment=true");
-      }
-      expect(url).toEqual(expectedUrl2);
-      expect(mockLocationAssign).toHaveBeenCalledWith(expectedUrl2);
-    });
+  it.each(validSkylinkVariations)("should download with a skylink url with attachment set", (fullSkylink) => {
+    mockLocationAssign.mockClear();
+    const url = client.downloadFile(fullSkylink);
+
+    const path = extractNonSkylinkPath(fullSkylink, skylink);
+
+    let expectedUrl2 = `${expectedUrl}${path}${attachment}`;
```

Comment:
just call it `fullExpectedUrl`, `expected` or something like that, adding an number to a variable because the name is already taken is not really descriptive